### PR TITLE
a11y: use onGetErrorMessage in createDialogModal

### LINKED ISSUE #2060: [Accessibility][Usable - New Dialog] : Once the error appears it can't remove again even after the removal of the filed which have error.
**User Experience** 
The screen reader dependent user will get confused as well as it effects low vision user as the error message didn't disappear even after correcting the field where the error appears.

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
Screen Reader: Narrator
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](https://github.com/microsoft/BotFramework-Composer)

**Repro Steps:** 
1. Open the above mentioned URL in EDGE.
2. Select 'Design Flow' from left navigation.
3. Now, navigate to 'New Dialog' button available in the filter dialog pane and activate it..
4. Type the name of the dialog which is already created ex- Actions.
5. Observer the issue after correcting the name of the dialog.

**Actual:** 
The error message i.e. "Duplication of the dialog" is not gets disappeared even after correcting the name in the dialog.

**Expected:**
As the user delete the previously entered dialog name the error message should gets disappeared.

**MAG Guideline:** [MAG](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/Doc.aspx?sourcedoc=%7BF98F4235-B6BD-47D3-8DA4-F28B9B6FBFA8%7D&file=Guideline%20101%20%E2%80%93%20Recommended%20Assistive%20Technology%20Tools%20for%20MAS%20Testing%20of%20Microsoft%20Digital%20Properties.docx&action=default&mobileredirect=true)

**Attachment:**
[Usable_error message can't gets corrected even after correcting the error field.zip](https://github.com/microsoft/BotFramework-Composer/files/4241282/Usable_error.message.can.t.gets.corrected.even.after.correcting.the.error.field.zip)

### LINKED ISSUE #2074: [Accessibility][Keyboard -New Dialog]: Error should displayed as soon as the mandatory field are blank or duplication.
**User Experience** 
User with vision disability will not get any information on clicking the next button then they need to navigate back to the 'Name" edit field which consumes unnecessarily increases the user's effort. 

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
Screen Reader: Narrator
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](https://github.com/microsoft/BotFramework-Composer)

**Repro Steps:** 
1. Open the above mentioned URL in EDGE.
2. Select 'Design Flow' from left navigation.
3. Now, navigate to 'New Dialog' button available in the filter dialog pane and activate it..
4. Type the name of the dialog which is already created ex- Actions or leave it blank and navigate forward to click next button.
5. Observer the issue.

**Actual:** 
As the user navigate through the mandatory field and leave it blank no error message displayed until they press next.

**Expected:**
On leaving the mandatory field blank or duplication of the name, the error message should be displayed as soon as they press 'Tab' key.

**MAS Impacted:** [MAS3.3.4](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={5f773cee-c1c7-4292-b330-cb2f4aad00c4})

**Attachment:**
[MAS3.3.4_Error should displayed as soon as the mandatory field.zip](https://github.com/microsoft/BotFramework-Composer/files/4249129/MAS3.3.4_Error.should.displayed.as.soon.as.the.mandatory.field.zip)

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This reworks createDialogModal.tsx a bit by moving the error checking into the Fabric-supported "onGetErrorMessage" and using a boolean in the modal's state to keep track of if we see any errors.

## Task Item

Closes #2074
Closes #2060 

## Screenshots

image

### CODE REVIEW

File: Composer/packages/client/src/pages/design/createDialogModal.tsx [hunk 1]
```diff
@@ -43,37 +43,35 @@ export const CreateDialogModal: React.FC<CreateDialogModalProps> = props => {
...
   };
 
-  const nameRegex = /^[a-zA-Z0-9-_.]+$/;
-  const validateForm = (data: DialogFormData) => {
-    const errors: { name?: string } = {};
-    const { name } = data;
-
+  const _onGetErrorMessage = (name: string) => {
     if (name) {
       if (!nameRegex.test(name)) {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/design/createDialogModal.tsx]: This is used before its defined.

**Comment by a-b-r-o-w-n** [comment]
This is actually addressing the same issue as #2338. So we may not need to overcomplicate with a hook, but if we want a consistent behavior for other forms, it seems worth the investment to me.

**Comment by beyackle** [comment]
> Can we just move the validation to the `onChange` handler?

That was the first thing I tried, and it didn't do the validation when I expected it should. This way, we're using the supported method from Fabric's own API documentation.

File: Composer/packages/client/src/pages/design/createDialogModal.tsx (hunk 1 above)
Comment (beyackle) [reply Composer/packages/client/src/pages/design/createDialogModal.tsx]: It's defined on line 46 and used on line 88. I don't see the issue.

**Comment by a-b-r-o-w-n** [comment]
Closing in favor of #2369

**Pull request closed without merging**